### ISSUE TITLE
FluentHub crashes after oauth callback

### ISSUE THREAD

**Issue by HenryWu01** [fluenthub-community/FluentHub#123]
### Before you start...

- [X] Have you updated the FluentHub? You might be using an old version.
- [X] Have you checked if similar issue have already been reported?

### What windows are you seeing the problem on?

Windows 11

### What FluentHub version/architecture are you seeing the problem on?

It's the newest from the default branch, I could not get into about page since login is required.

### Description

![image](https://user-images.githubusercontent.com/99880210/164714028-746fb90c-e76c-46e9-9932-14230c238980.png)
![image](https://user-images.githubusercontent.com/99880210/164714086-994a1005-fa0a-46e7-836c-bdd034843f96.png)
![image](https://user-images.githubusercontent.com/99880210/164714721-70069408-48e0-45b8-b25d-25b29a18a3d7.png)

### Steps To Reproduce

1.Start setup
2.Follow to Github and complete authorization
3.Callback to FluentHub and crash

### Expected behavior

No crash, successfully got into the main page.

**Comment by onein528** [comment]
can you give us log file?(`C:\Users\USERNAME\AppData\Local\Packages\FluentHubDev_9es9q7t8fr9b8\LocalState`)

**Comment by HenryWu01** [comment]
Sure, here is the log.
[Log20220422.log](https://github.com/fluenthub-community/FluentHub/files/8541316/Log20220422.log)

**Comment by onein528** [comment]
it's unclear little bit. `An error occurred while sending the request` this error happen somethimes with no internet. but I am tracking this issue.

**Comment by onein528** [comment]
I am adding log more to make sure know where error or exception occured in now branch, so can you give me log again after merged that branch?

**Comment by HenryWu01** [comment]
Can you give me the branch name? I did not found it

**Comment by onein528** [comment]
I'm working on issue/PR event getters, so I will add on this: `issuepr-events` branch.

**Comment by onein528** [comment]
added now first commit about this. can you check out, build, and, give me log?

**Comment by HenryWu01** [comment]
Sure, I will check it out now.

**Comment by HenryWu01** [comment]
Here is the log.
[Log20220422.log](https://github.com/fluenthub-community/FluentHub/files/8541675/Log20220422.log)

**Comment by onein528** [comment]
ah, I cannot fix right away. it takes long time to fix. sorry but this is first priority!

**Comment by onein528** [comment]
just checking, can you build with two breakpoint?

here
![image](https://user-images.githubusercontent.com/62196528/164732762-5f157c3e-ff2f-4891-b181-70964db8ccac.png)

maybe your machine through first breakpoint but please check these value not null

![image](https://user-images.githubusercontent.com/62196528/164733082-f5a74457-465d-4cf7-85a3-d4d18aa21cbb.png)

**Comment by HenryWu01** [comment]
Sure, please wait a minute.

**Comment by HenryWu01** [comment]
![image](https://user-images.githubusercontent.com/99880210/164733848-d2ca7c16-7bd9-41e4-b3e5-6ad71f66783a.png)
No, there all have value

**Comment by onein528** [comment]
ohh very compacated 🤯. confusing...

**Comment by onein528** [comment]
do you have here's?

![image](https://user-images.githubusercontent.com/62196528/164734656-845a6ba0-f48a-4705-97d9-53c53b1d64d0.png)

**Comment by HenryWu01** [comment]
No, I only have this one:
![image](https://user-images.githubusercontent.com/99880210/164734945-bc2bb8c4-d0b7-4b3e-b69b-1baabb19e040.png)

**Comment by onein528** [comment]
it seems that method fails when creating access token.

**Comment by HenryWu01** [comment]
Wired...I will have a look too

**Comment by HenryWu01** [comment]
You're right, it crashes when executing ``CreateAccessToken`` function.

**Comment by HenryWu01** [comment]
The error message is ``安全频道支持出错``, which I think might be related to TLS version?

**Comment by onein528** [comment]
FYI https://stackoverflow.com/questions/21354992/an-error-occurred-in-the-secure-channel-support-classic-asp-http-request

maybe true and is not FluentHub problem. this is caused in HttpClient

**Comment by HenryWu01** [comment]
Might be related to .NET's HttpClient, Windows 11 or OctoKit, got not idea to resolve this issue. All information available is for VB, ASP, etc.

**Comment by HenryWu01** [comment]
Do you have like a release version of FluentHub? I want to try that out to eliminate some possibilities.

**Comment by BobbyESP** [comment]
> Do you have like a release version of FluentHub? I want to try that out to eliminate some possibilities.

I'm looking for a Release build, wait a minute please

**Comment by BobbyESP** [comment]
https://drive.google.com/file/d/1nvdwftYfFkBGRZctieVIK2rPfQ4Qct_l/view?usp=sharing 

Please, try with that one. Build date: 2/4/22

**Comment by onein528** [comment]
we can create on azure pipelines. im on it

**Comment by onein528** [comment]
https://dev.azure.com/fluenthub-community/FluentHub/_build/results?buildId=598&view=results

when it finished, click 'artifacts' and can download packages

**Comment by BobbyESP** [comment]
Nice

**Comment by HenryWu01** [comment]
Thank you. Unfortunately it crashes again :|

**Comment by HenryWu01** [comment]
Here is the new log:
[Log20220423.log](https://github.com/fluenthub-community/FluentHub/files/8545809/Log20220423.log)

I think we may create a issue in Octokit?

**Comment by HenryWu01** [comment]
It turns out that my hypothesis was actually correct, for those who face this issue, open "Internet Properties", click "Advance" and Select all TLS and SSL version. This even solves My Microsoft Store issue! Weird Windows 11.

**Comment by HenryWu01** [comment]
Like this:
![image](https://user-images.githubusercontent.com/99880210/164863685-27770148-4c68-4920-bf87-8c0dd2b0272f.png)

**Comment by onein528** [comment]
nice. did you fix this issue?

**Comment by HenryWu01** [comment]
> nice. did you fix this issue?

Yes, After I change the SSL & TLS version, this issue immediately went away. Even my Microsoft Store went to normal

**Comment by onein528** [comment]
do you think this issue closed? you had to change that settings but I think this problem is not related our app.

**Comment by HenryWu01** [comment]
Sure, this is issue is not related to FluentHub at all. Just some weird Windows 11 issue.

**Comment by BobbyESP** [comment]
Nice! We should add this information to the README, or the installation guide

**Comment by onein528** [comment]
good idea, this app have some restrictions.

**Comment by nathayoung** [comment]
@HenryWu01 are you running the GA version of Win 11 or are you running an insider build?  I still can't get this to work but I'm in the Win 11 Dev ring so I'm wondering if I can blame that...

**Comment by BobbyESP** [comment]
@nathayoung  Please, go to Win + R and write inetcpl.cpl. After that, go to the advanced settings and scroll all down of that menu. Now, activate all TLS and SSL versions, restart the computer if is needed and try to login.

**Comment by BobbyESP** [comment]
<img width="79" alt="image" src="https://user-images.githubusercontent.com/60316747/165113319-790574fe-6bf0-4170-a48f-fd92e5652318.png">

**Comment by onein528** [comment]
We need notice this. I created General Updates 7 branch and I will change README

**Comment by onein528** [comment]
how about?

![image](https://user-images.githubusercontent.com/62196528/165115515-98e0aaea-6009-42f6-9360-f9d6d1a323ef.png)

**Comment by BobbyESP** [comment]
instead of enabled, disabled.

**Comment by BobbyESP** [comment]
Because the problem is when they're off. But nice!

**Comment by BobbyESP** [comment]
Nice @onein528

**Comment by hedhehd** [comment]
Did not help!

**Comment by hedhehd** [comment]
Even a restart

**Comment by BobbyESP** [comment]
Wow, that's truly weired.

**Comment by hedhehd** [comment]
0.7.13

**Comment by BobbyESP** [comment]
Ok, thanks. Did you readed the how to build reamde?

**Comment by hedhehd** [comment]
Yes

**Comment by hedhehd** [comment]
Did exactly the same

**Comment by hedhehd** [comment]
Wait there's a log file?

**Comment by onein528** [comment]
Can you authoriza the app(and then, You can go homepage even if whatever occur)?

**Comment by hedhehd** [comment]
How

> Can you authoriza the app(and then, You can go homepage even if whatever occur)?

How?

**Comment by onein528** [comment]
just launch app.

**Comment by onein528** [comment]
![image](https://user-images.githubusercontent.com/62196528/165317412-ecd16565-c411-4509-be2d-809b93efe2f6.png)

**Comment by onein528** [comment]
Isee... Did you put credentials to config file?

**Comment by hedhehd** [comment]
Yes

**Comment by hedhehd** [comment]
Or else #129

**Comment by onein528** [comment]
@hedhehd 
![image](https://user-images.githubusercontent.com/62196528/165319281-971bc42c-8c15-4b9c-bc02-df93755f884a.png)

can you set breakpoints in these point?

**Comment by onein528** [comment]
and get values in `query` valuable(hover `query` and get `.Query` value).

List this:

![image](https://user-images.githubusercontent.com/62196528/165320464-bc0931ac-6dd2-461a-99c3-ac8ea922d1d6.png)

**Comment by hedhehd** [comment]
> and get values in `query` valuable(hover `query` and get `.Query` value).
> 
> List this:
> 
> ![image](https://user-images.githubusercontent.com/62196528/165320464-bc0931ac-6dd2-461a-99c3-ac8ea922d1d6.png)

I don't know how even if you said it.

**Comment by hedhehd** [comment]
This is vs 2022

**Comment by hedhehd** [comment]
Which vs do I have to use?

**Comment by onein528** [comment]
yes, 2022. 

1. set breakpoints
2. build
3. get value

**Comment by hedhehd** [comment]
Breakpoints can't be set there

**Comment by onein528** [comment]
i will sleep

**Comment by hedhehd** [comment]
Okay

**Comment by nathayoung** [comment]
> ![image](https://user-images.githubusercontent.com/81228260/165317618-7165ab5e-2ebf-4bcf-b7fa-5cae404dc995.png) It stay's like this for 5 seconds and then it crashes

I can replicate this.  I'll send over what I have

**Comment by hedhehd** [comment]
IT WORKED

**Comment by hedhehd** [comment]
I UPDATED IT

**Comment by hedhehd** [comment]
Issue is closed

**Comment by onein528** [comment]
very heated. @BobbyESP Can we lock this conversation?

**Comment by onein528** [comment]
or we can create new conversation.

**Comment by onein528** [comment]
@nathayoung App does work in your machine?

**Comment by nathayoung** [comment]
> @nathayoung App does work in your machine?

just fetched.  on the main branch and not getting any new stuff.  I even moved to an entirely new machine and started completely over.  This is crazy 😂

**Comment by onein528** [comment]
okay.. do you get issue even though?

**Comment by nathayoung** [comment]
Yup.  Just installed it on my machine.  is this what you want: 
![image](https://user-images.githubusercontent.com/46688736/165334776-d11f01be-c19b-4ea3-aa75-7016b7245b41.png)

**Comment by onein528** [comment]
Nice. it's up-to-date. can you build and launch?

**Comment by nathayoung** [comment]
I'll send a video shortly

**Comment by nathayoung** [comment]
Here's what's happening for me... 

https://user-images.githubusercontent.com/46688736/165337396-7b99a6a7-afaa-4c15-9638-a471a91e574d.mp4

**Comment by onein528** [comment]
What.. very confusing. can you track with setting breakpoint?

**Comment by nathayoung** [comment]
I can but first it gave me a suggested action to install System.Runtime. I might have missed an installation on setup.  Let me try again with that installed.

That worked.   Finally!  Thanks for the support!  I apologize for my incompetence 😂
![image](https://user-images.githubusercontent.com/46688736/165339579-562c7f63-aaec-4800-99c5-720f591c4b04.png)

**Comment by onein528** [comment]
And if you want to unlock/reopen, let us know in discord.
